src/main/scala/intellij/haskell/action/CreateHaskellFileAction.scala

```diff
@@ -58,48 +58,54 @@ class CreateHaskellFileAction extends CreateFileFromTemplateAction(CreateHaskell
   }
 
   override def createFileFromTemplate(originName: String, template: FileTemplate, originDir: PsiDirectory): PsiFile = {
-    val pathItems = HaskellFileUtil.getPathFromSourceRoot(originDir.getProject, originDir.getVirtualFile).orNull
-
-    // Adapted from super definition.
-    val mkdirs = new CreateFileAction.MkDirs(originName, originDir)
-    val name = mkdirs.newName
-    val dir = mkdirs.directory
-    val project = dir.getProject
-
-    val nameWithmodulePrefix = if (pathItems == null || invalidPathItems(pathItems) || pathItems.isEmpty) {
-      name
-    } else {
-      pathItems.mkString(".") + "." + name
-    }
+    val pathItems = HaskellFileUtil.getPathFromSourceRoot(originDir.getProject, originDir.getVirtualFile)
```

Comment:
I see, weird, with files it's safer to use getContainingFile. The PsiDirectoryImpl just returns null...